Add word case option to passphrase generator

shaneknysh opened the pull request:
adds lower, upper, and title case options to passphrase (diceware) generation

Adds an option to have passphrases generated in lower, UPPER, or title case. Adds this option to the saved configuration.

## Type of change
- ✅ New feature (non-breaking change which adds functionality)

## Description and Context
No large changes, ads a small step in teh gerenation of passphrases that modifies the word selected form the word list based on desired capitalization

fixes #1933

## Screenshots
![image](https://user-images.githubusercontent.com/158361/58110655-34636a80-7bad-11e9-9bf6-7f48c3195d00.png)

## Testing strategy
Manual testing only. I ran the compiled executable and generated a new passphrase with the new options. The passphrases were all generated with the correct case based on the selected option. 

I opened the executable multiple time and the case option was saved between sessions.

Repeated all manual tests with the standalone password generator.

No major documentation changes required - added comments for 0 = lower, 1 = UPPER, and 2 = Title in several locations in the code to make the values clear. Considered an enum but decided to go with numbers. 

## Checklist:
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**

- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**
I built using the release tool - is this accomplished with the release tool or do I have to do a manual step?

- ✅ My change requires a change to the documentation, and I have updated it accordingly.
- ✅ I have added tests to cover my changes.
I did not find any tests that needed to be updated and I am not certain if any need to be added. I am open to creating new tests if necessary I just need some guidance as to what is required.

droidmonkey commented on src/core/PassphraseGenerator.cpp:
```diff
@@ -98,7 +109,23 @@ QString PassphraseGenerator::generatePassphrase() const
+
+        //convert case
+        switch (m_wordCase) {
+        case 0 : // lower case
```
Prefer not to use "magic numbers" instead define an enum in the class:
```
enum PassphraseCase
{
	LOWERCASE,
	UPPERCASE,
	TITLECASE
};
```

shaneknysh replied:
changes made

droidmonkey commented on src/core/PassphraseGenerator.cpp:
```diff
@@ -98,7 +109,23 @@ QString PassphraseGenerator::generatePassphrase() const
+        case 2 : // Title Case
+            tmpWord = tmpWord.replace(0,1,tmpWord.left(1).toUpper());
+            break;
+        default :
```
The default should mimic LOWERCASE and do toLower

shaneknysh replied:
changes made

shaneknysh commented:
I will make the changes tonight/ tomorrow. 

I may need some guidance / advice on the enum. I had an enum declared in PassphraseGenerator.h in the public scope but the enum was not available in PasswordGeneratorWidget.cpp. I'm not certain what I was doing wrong and I did not want to declare the enum twice.

droidmonkey commented:
See here for an example: https://github.com/keepassxreboot/keepassxc/blob/develop/src/core/Entry.h#L159

shaneknysh commented:
I've updated to make the requested changes. I also added three new tests. 

I would love some advice on running the test suite. I've tried under windows and Ubuntu 18 and in both cases I get the error 
`make: *** No rule to make target 'test'.  Stop.`

droidmonkey commented:
I cleaned up the code a bit and changed the radio buttons to a combo box. Ready for merge.

![image](https://user-images.githubusercontent.com/2809491/58338377-bf469e00-7e15-11e9-98bc-9259f88a4cea.png)

shaneknysh commented:
I am not an expert in Qt and haven't written c++for production in a decade. 

I am a UX designer and a combo box for 2-3 choices in a worse choice from a UX point of view. Unless you expect more choices being added to the word case list I would recommend sticking with the combo boxes.

droidmonkey commented:
I tend to agree with you on the radio buttons vs combobox.... but not when the options are laid out horizontally. It just looked and felt awkward. The combobox is neat and tidy and won't get in the way if we add other options to the passphrase generator (custom wordlists) in the future.
